operator/cmd/gke_addon_poststart/main.go

```diff
@@ -49,22 +64,31 @@ var configConnectorResource = schema.GroupVersionResource{
 // after the manager container is created. This is meant to be used for the GKE
 // add-on only, not the standalone operator.
 func main() {
+	var managerNamespaceSuffix string
+	flag.StringVar(&managerNamespaceSuffix, "manager-namespace-suffix", "", "Create controller manager pod/SA in a separate namespace, replacing suffix of watched namespace with the specified suffix.")
+	flag.Parse()
+
 	ctx := context.Background()
 	dynamicClient, err := dynamic.NewForConfig(ctrl.GetConfigOrDie())
 	if err != nil {
 		log.Fatalf("error creating dynamic client: %v", err)
 	}
-	if err := createDefaultConfigConnector(ctx, dynamicClient); err != nil {
+	if err := createDefaultConfigConnector(ctx, dynamicClient, managerNamespaceSuffix); err != nil {
 		log.Fatalf("error creating default ConfigConnector object: %v", err)
 	}
 }
 
 // createDefaultConfigConnector creates a ConfigConnector object on the K8s API
 // server. This is done for users who want to have a default ConfigConnector
 // object created for them upon enabling the GKE add-on.
-func createDefaultConfigConnector(ctx context.Context, dynamicClient dynamic.Interface) error {
+func createDefaultConfigConnector(ctx context.Context, dynamicClient dynamic.Interface, managerNamespaceSuffix string) error {
 	u := &unstructured.Unstructured{}
-	b := []byte(defaultConfigConnector)
+	var b []byte
+	if managerNamespaceSuffix == "" {
+		b = []byte(defaultConfigConnector)
+	} else {
+		b = []byte(fmt.Sprintf(defaultConfigConnectorNamespaceSeparated, managerNamespaceSuffix))
+	}
```

Comment:
Suggest making this a Unstructured level operations, especially considering that a string formatting is used. 
```go
u.SetLabels(
map[string]string{
  ...
}
)
```

Reply:
Done